File: kotlin_java/src/baekjoon/silver/s2/DFS와_BFS.java
package baekjoon.silver.s2;

import java.util.LinkedList;
import java.util.Queue;
import java.util.Scanner;

public class DFS와_BFS {
    static int[][] graph;
    static boolean[] visited;
    static int n;
    static int m;
    static int startPoint;

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        n = sc.nextInt();
        m = sc.nextInt();
        startPoint = sc.nextInt();

        graph = new int[1001][1001];
        visited = new boolean[1001];

        for(int i = 0; i<m; i++) {
            int x = sc.nextInt();
            int y = sc.nextInt();

            graph[x][y] = graph[y][x] = 1;
        }

        dfs(startPoint);
        visited = new boolean[1001];
        bfs();
    }

    static void dfs(int p) {
        visited[p] = true;
        System.out.print(p+" ");

        for(int j = 1; j<=n; j++) {
            if(graph[p][j] == 1 && visited[j] == false) {
                dfs(j);
            }
        }
    }

    static void bfs() {
        System.out.println();

        Queue<Integer> queue = new LinkedList<Integer>();
        queue.add(startPoint);
        visited[startPoint] = true;
        System.out.print(startPoint + " ");

        while (queue.isEmpty() == false) {
            int temp = queue.poll();

            for(int j = 1; j<=n; j++) {
                if(graph[temp][j] == 1 && visited[j] == false) {
                    queue.add(j);
                    visited[j] = true;
                    System.out.print(j+" ");
                }
            }
        }
    }
}

File: src/main/java/baekjoon/silver/s2/DFS와 BFS.kt
package baekjoon.silver.s2

import java.util.*

var graph: Array<IntArray>? = null
var visited: BooleanArray? = null
var n = 0
var m = 0
var startPoint = 0

fun main() {
    val sc = Scanner(System.`in`)
    n = sc.nextInt()
    m = sc.nextInt()
    startPoint = sc.nextInt()
    graph = Array(1001) { IntArray(1001) }!!
    visited = BooleanArray(1001)
    for (i in 0 until m) {
        val x = sc.nextInt()
        val y = sc.nextInt()
        graph!![y][x] = 1
        graph!![x][y] = graph!![y][x]
    }
    dfs(startPoint)
    visited = BooleanArray(1001)
    bfs()

}

fun dfs(p: Int) {
    visited!![p] = true
    print("$p ")
    for (j in 1..n) {
        if (graph!![p][j] == 1 && !visited!![j]) {
            dfs(j)
        }
    }
}

fun bfs() {
    println()
    val queue: Queue<Int> = LinkedList()
    queue.add(startPoint)
    visited!![startPoint] = true

    print("$startPoint ")

    while (queue.isNotEmpty()) {
        val temp = queue.poll()
        for (j in 1..n) {
            if (graph!![temp][j] == 1 && !visited!![j]) {
                queue.add(j)
                visited!![j] = true
                print("$j ")
            }
        }
    }
}